### PR TITLE
Add TLS support for scheduler.

### PR DESCRIPTION
**Pull request by qibobo**

### CODE REVIEW

**Comment by fraenkel** [comment]
Where are the changes for the scheduler itself?

**Comment by qibobo** [comment]
Add TLS support for scheduler

File: scheduler/pom.xml [hunk 1]
```diff
@@ -91,6 +91,12 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-actuator</artifactId>
 		</dependency>
+		<!-- https://mvnrepository.com/artifact/org.apache.httpcomponents/httpclient -->
+		<dependency>
+		    <groupId>org.apache.httpcomponents</groupId>
```
Comment (fraenkel) [inline scheduler/pom.xml]: why?

File: scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/ssl/CustomClientHttpRequestFactory.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+
+public class CustomClientHttpRequestFactory extends HttpComponentsClientHttpRequestFactory {
```
Comment (fraenkel) [inline scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/ssl/CustomClientHttpRequestFactory.java]: why?
We should just support HTTPS which means we just configure application.properties and that is it.

File: scheduler/src/main/resources/application.properties [hunk 3]
```diff
@@ -12,3 +12,11 @@ scalingengine.notification.reschedule.maxcount=3
...
 # scaling engine url
 autoscaler.scalingengine.url=http://test-scalingengine-test
+
+#ssl
+server.ssl.key-store=../test-certs/scheduler.jks
+caCert=../test-certs/autoscaler-ca.crt
+server.ssl.key-alias=scheduler
+server.ssl.key-store-password=123456
+#server.ssl.key-password=123456
+server.ssl.key-store-type=JKS
```
Comment (fraenkel) [inline scheduler/src/main/resources/application.properties]: why not just use pkcs12? makes it easier for everyone. which is what you generated from the tooling.

**Comment by qibobo** [comment]
remove useless code.

**Pull request merged**